src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3304,6 +3298,7 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
                 if (initializerArgumentListOpt != null)
                 {
                     this.BindArgumentsAndNames(initializerArgumentListOpt, diagnostics, analyzedArguments, allowArglist: true);
+                    diagnostics = analyzedArguments.AvoidCascading(diagnostics);
```

Comment:
Is `AvoidCascading()` necessary? Isn't there a good chance that subsequent diagnostics reported by this method are independent of errors binding arguments?